node_modules/oae-admin/tenants/js/tenants.js

```diff
@@ -350,43 +309,65 @@ define(['jquery', 'oae.core', 'jquery.jeditable'], function($, oae) {
+            var url = '/api/search/tenants?disabled=true';
+            if (History.getState().data.query) {
+                url += '&q=' + encodeURIComponent(History.getState().data.query);
+            }
 
-            var tenantsArr = [];
-            $.each(tenantsToRender, function(tenantHost, tenant) {
-                tenantsArr.push(tenant);
+            // Set up the infinite scroll for tenants
+            infinityScroll = $('#tenants-table-rows', $rootel).infiniteScroll(url, {'limit': 50}, $('#tenants-row-template', $rootel), {
+                'postProcessor': function(data) {
+                    tenantsCache = _.indexBy(data.results, 'alias');
+                    return _.extend({}, data, {'context': widgetData.context});
+                },
+                'emptyListProcessor': function() {
+                    oae.api.util.template().render($('#tenants-none-template', $rootel), null, $('#tenants-table-rows', $rootel));
+                },
+                'postRenderer': postRenderer
             });
+        };
 
-            // Sort the tenants alphabetically based on the tenant alias
-            tenantsArr.sort(function(a, b) {
-                if (a.alias > b.alias) {
-                    return 1;
-                } else if (a.alias < b.alias) {
-                    return -1;
-                } else {
-                    return 0;
-                }
-            });
+        /**
+         * Render the list of tenants
+         */
+        var renderTenants = function() {
+            oae.api.util.template().render($('#tenants-template', $rootel), {'context': widgetData.context}, $('#tenants-container', $rootel));
+
+            // Get the current search query from the History.js data object
+            var query = History.getState().data.query;
+            $('#tenants-search-query').val(query);
+
+            // Only render a single tenant when we're looking at a specific tenant
+            if (widgetData.context.isGlobalAdminServer) {
+                setUpInfiniteScroll(function(data, $templateOutput) {
+                    enableInlineEdit($templateOutput);
+                });
+            } else {
+                renderTenant(widgetData.context);
+                enableInlineEdit($('#tenants-table'));
+            }
 
-            oae.api.util.template().render($('#tenants-template', $rootel), {
-                'tenants': tenantsArr,
-                'hasStoppedServer': hasStoppedServer,
-                'context': widgetData.context
-            }, $('#tenants-container', $rootel));
-            enableInlineEdit();
             // Set up the validation for the create tenant form
             oae.api.util.validation().validate($('#tenants-create-form', $rootel), {'submitHandler': createTenant});
+
+            // When the search tenants form is submit, we intercept and refresh the infinite scroll
+            // with the results
+            $('#tenants-search > form', $rootel).on('submit', function() {
```

Comment:
Isn't this going to bing multiple times as the `renderTenants` will be executed every time a new tenant is created